### ISSUE TITLE
Deadlock solving graph using proposed release IIF 0.8.3 and DFG 0.5.2 

### ISSUE THREAD

**Issue by GearsAD** [JuliaRobotics/IncrementalInference.jl#485]
It looks like it's deadlocking on a test graph? This is a reproducible sample:

```julia
using Logging
using DistributedFactorGraphs
using IncrementalInference
logger = SimpleLogger(stdout, Logging.Debug)
global_logger(logger)
dfg = initfg() #LightDFG{SolverParams}(solverParams=SolverParams())
# Add some nodes.
v1 = addVariable!(dfg, :a, ContinuousScalar, labels = [:POSE])
v2 = addVariable!(dfg, :b, ContinuousScalar, labels = [:POSE])
v3 = addVariable!(dfg, :c, ContinuousScalar, labels = [:LANDMARK])
f1 = addFactor!(dfg, [:a; :b], LinearConditional(Normal(50.0,2.0)) )
f2 = addFactor!(dfg, [:b; :c], LinearConditional(Normal(50.0,2.0)) )
tree, smtasks, hist = solveTree!(dfg)
```

Travis: https://travis-ci.org/JuliaRobotics/DistributedFactorGraphs.jl/jobs/626015919

**Comment by dehann** [comment]
right, will have a look thanks!

**Comment by dehann** [comment]
assuming these are all current master on DFG and IIF?

**Comment by dehann** [comment]
also, if this fails so easy we should perhaps add this as standard test in IIF, not via loopback into special DFG downstream tests

**Comment by dehann** [comment]
Oh, but this has no prior -- so the answers wont be correct in this test regardless.

**Comment by dehann** [comment]
yes, the test fails by me too with CSM cycling indefinitely -- will need to resolve this before we can tag either DFG or IIF thanks

**Comment by dehann** [comment]
This is likely a consequence of changes relating to #474.

**Comment by GearsAD** [comment]
Yep cycle loopback isn't great for finding these, will add as test in IIF :+1:

**Comment by dehann** [comment]
without prior, here is the current Bayes tree:
![Screenshot from 2019-12-18 00-54-28](https://user-images.githubusercontent.com/6412556/71059532-194fdd80-2131-11ea-8a4d-75e3db127e2a.png)

**Comment by dehann** [comment]
this tree is wrong in that one of the factors (bcf1) looks to be missing.

**Comment by dehann** [comment]
So this could also just be a visualization mistake, because if I check manually the factor is there:
![Screenshot from 2019-12-18 08-47-56](https://user-images.githubusercontent.com/6412556/71091678-8edc9d80-2173-11ea-95a1-614f1eb4e29d.png)

**Comment by dehann** [comment]
likely related to #461

**Comment by dehann** [comment]
One of the problems here is likely the same thing as in #458 .

There is a separate visualization issue where leaf clique is not showing the factors that are in the clique -- will resolve that in a separate issue #489.

**Comment by dehann** [comment]
test includes prior and visualization issue tracked separately, therefore closing this:
- https://github.com/JuliaRobotics/IncrementalInference.jl/blob/52b44d800073026ea687bd051e6659fc2cfc801c/test/testBasicCSM.jl#L28
